core/store/tx_manager.go

```diff
@@ -741,6 +788,12 @@ func (a *ManagedAccount) GetAndIncrementNonce(callback func(uint64) error) error
 	return err
 }
 
+func (a *ManagedAccount) updateLastConfirmedNonce(latest uint64) {
+	if latest > a.lastConfirmedNonce {
+		a.lastConfirmedNonce = latest
+	}
```

Comment:
This seems like a program invariant, in that the code has been designed so the else case should never happen - perhaps a log Error is warranted here in case something weird happens

Reply:
It is possible during a reorg. Maybe just warn that it was seen?

Reply:
Ok, converted this to only update when a transaction is marked as safe. Should avoid reorg issues, or at least give the operator the option of how many confirmations they require.